src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -531,6 +543,29 @@ std::vector<MessagePtr> parseUserNoticeMessage(Channel *channel,
         {
             messageText = "Announcement";
         }
+        else if (msgType == "raid")
+        {
+            auto loginTag = tags.find("login");
+            auto displayNameTag = tags.find("msg-param-displayName");
```

Comment:
Since we don't care to make different implementation depending on if the value is unset, or if the value is set, or if the value is set but empty, instead of doing the `v = tags.find("..."); v != tags.end();` dance, use `tags.value("...").toString()`, then check if the string is valid/non-empty instead.
Same goes for the color code, that way we don't need to blindly trust that it's a valid color, and keep the SystemColor if the color is invalid

Reply:
Done in [c620474](https://github.com/Chatterino/chatterino2/pull/5651/commits/c6204743a66916eb7ad9439ad8d490dbcd308bad) and [61f77a1](https://github.com/Chatterino/chatterino2/pull/5651/commits/61f77a1beb46878e946212b73d70555f0e647aea)